app/controllers/organizations/invitations_controller.rb

```diff
@@ -3,29 +3,50 @@ class Organizations::InvitationsController < Devise::InvitationsController
 
   layout "dashboard", only: [:new, :create]
 
+  # Do not use this action! Use other invitation controllers instead for new.
   def new
-    authorize! StaffAccount, context: {organization: Current.organization},
+    authorize! User, context: {organization: Current.organization},
       with: Organizations::InvitationPolicy
 
     @user = User.new
-    @staff = StaffAccount.new(user: @user)
   end
 
   def create
-    authorize! StaffAccount, context: {organization: Current.organization},
-      with: Organizations::InvitationPolicy
-
-    @user = User.new(
-      user_params.merge(password: SecureRandom.hex(8)).except(:roles)
-    )
-    @user.add_role(user_params[:roles], Current.organization)
-    @user.staff_account = StaffAccount.new
-
-    if @user.save
-      @user.invite!(current_user)
-      redirect_to staff_index_path, notice: "Invite sent!"
+    case user_params[:roles]
```

Comment:
We use a case statement here to account for the different authorization paths necessary for the action as well as the separate results.

Fosterers can be invited by staff members but only an admin can invite staff and other admins.